android/src/main/java/com/onegini/mobile/sdk/reactnative/OneginiSDK.kt

```diff
@@ -48,54 +42,34 @@ class OneginiSDK @Inject constructor(
         val clientBuilder = OneginiClientBuilder(applicationContext, createPinRequestHandler, pinAuthenticationRequestHandler)
 
         clientBuilder.setBrowserRegistrationRequestHandler(registrationRequestHandler)
-                .setHttpConnectTimeout(Constants.httpConnectTimeoutBrowserRegistrationMiliseconds)
-                .setHttpReadTimeout(Constants.httpReadTimeoutBrowserRegistrationMiliseconds)
-
-        setProviders(clientBuilder)
-        setConfigModel(clientBuilder)
-        setSecurityController(clientBuilder)
+            .setMobileAuthWithOtpRequestHandler(mobileAuthOtpRequestHandler)
+            .setFingerprintAuthenticationRequestHandler(fingerprintAuthenticationRequestHandler)
+            .setHttpConnectTimeout(Constants.httpConnectTimeoutBrowserRegistrationMiliseconds)
+            .setHttpReadTimeout(Constants.httpReadTimeoutBrowserRegistrationMiliseconds)
 
-        if (config.enableMobileAuthenticationOtp) {
-            clientBuilder.setMobileAuthWithOtpRequestHandler(mobileAuthOtpRequestHandler)
-        }
+        val identityProvider = loadIdentityProvidersFromConfig(context)
+        addIdentityProviders(identityProvider, clientBuilder)
+        return clientBuilder.build()
+    }
 
-        if (config.enableFingerprint) {
-            clientBuilder.setFingerprintAuthenticationRequestHandler(fingerprintAuthenticationRequestHandler)
+    private fun loadIdentityProvidersFromConfig(context: Context): List<ReactNativeIdentityProvider> {
+        return try {
+            val configClass = ClassLoader(context).getClassByName("ReactNativeConfig").newInstance() as OneginiReactNativeConfig
+            configClass.getIdentityProviders()
+        } catch (e: ClassNotFoundException) {
+            Log.e("Loading of ReactNativeConfig failed.", e.toString())
+            emptyList()
+        } catch (e: InstantiationException) {
+            Log.e("onegini", "Loading of ReactNativeConfig failed. $e")
+            emptyList()
         }
```

Comment:
Can't this be one generic `catch` block? For both exaception the error and action are the same